MeshLib/MeshEditing/projectMeshOntoPlane.h

```diff
@@ -0,0 +1,58 @@
+/**
+ * \file   projectMeshOntoPlane
+ * \author Karsten Rink
+ * \date   2015-04-10
+ * \brief  Definition of the projectMeshOntoPlane
+ *
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef PROJECTMESHONTOPLANE_H
+#define PROJECTMESHONTOPLANE_H
+
+#include <vector>
+
+#include "MathLib/MathTools.h"
+#include "MathLib/Point3d.h"
+#include "MathLib/Vector3.h"
+
+#include "MeshLib/Mesh.h"
+#include "MeshLib/Node.h"
+#include "MeshLib/MeshEditing/DuplicateMeshComponents.h"
+
+
+namespace MeshLib {
+
+/**
+ * Projects all nodes of a mesh onto a plane specified by a point of origin and a normal vector.
+ * Overlapping elements, collapsed nodes, and other issues are not handled by the method.
+ * The normal vector need not be normalized.
+ */
+MeshLib::Mesh* projectMeshOntoPlane(MeshLib::Mesh const& mesh, MathLib::Point3d const& plane_origin, MathLib::Vector3 const& plane_normal)
+{
+	std::size_t const n_nodes (mesh.getNNodes());
+	std::vector<MeshLib::Node*> const& nodes (mesh.getNodes());
+	MathLib::Vector3 normal (plane_normal);
```

Comment:
The copy can be omitted if an already normalized plane normal is given as argument.